OptimizelySDK/AudienceConditions/BaseCondition.cs

```diff
@@ -30,28 +32,41 @@ public class BaseCondition : ICondition
         /// </summary>
         public const string CUSTOM_ATTRIBUTE_CONDITION_TYPE = "custom_attribute";
 
+        [JsonProperty("type")]
         public string Type { get; set; }
 
+        [JsonProperty("match")]
         public string Match { get; set; }
 
+        [JsonProperty("name")]
         public string Name { get; set; }
 
+        [JsonProperty("value")]
         public object Value { get; set; }
 
-        public bool? Evaluate(ProjectConfig config, UserAttributes userAttributes)
+        public bool? Evaluate(ProjectConfig config, UserAttributes userAttributes, ILogger logger)
         {
             if (Type == null || Type != CUSTOM_ATTRIBUTE_CONDITION_TYPE)
+            {
+                logger.Log(LogLevel.WARN, $@"Audience condition ""{this}"" has an unknown condition type. You may need to upgrade to a newer release of the Optimizely SDK");
                 return null;
+            }
 
             if (!IsValidMatchType())
+            {
+                logger.Log(LogLevel.WARN, $@"Audience condition ""{this}"" uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK");
```

Comment:
Nit: we can get rid of this block (and the `IsValidMatchType` method, which is currently coupled to `GetEvaluator`) if we move this "unknown match type" down to the part where we check for `evaluator == null`.